src/http/client.cc

```diff
@@ -340,6 +340,22 @@ future<> client::make_request(request req, reply_handler handle, reply::status_t
     return do_with(std::move(req), std::move(handle), [this, expected] (request& req, reply_handler& handle) mutable {
         return with_connection([this, &req, &handle, expected] (connection& con) {
             return do_make_request(con, req, handle, expected);
+        }).handle_exception_type([this, &req, &handle, expected] (const transport_error& ex) {
+            if (ex.requests_submitted == 1) {
+                // It wasn't connection from pool, no need in making new one again
+                return make_exception_future<>(ex.cause);
```

Comment:
Why?
I don't understand :-(
If I understood your code correctly, when the pool was empty you created a new connection and we have requests_submitted==1. But what if this connection breaks in the middle - which is very possible? Why should we not retry it? Why do you only retry if the connection was recycled from the pool?

Reply:
Renamed ex.requests_submitted into ex.connection_from_pool and tuned up the comment
Also there's more elaborated description of this place above, so I'm resolving this particular comment